### PR TITLE
feat(experiments HogQL rewrite): prepare trend queries

### PR DESCRIPTION
**Pull request by jurajmajerik**
## Problem
Same as https://github.com/PostHog/posthog/pull/25096 but for Trend experiments.

On an experiment, the query is stored in a "raw" format, without the experiment date range or feature flag breakdowns. This allows users to preview the goal in the UI and see the event frequency from the past, where the experiment date range and breakdowns aren't relevant.

## Changes
- The old implementation that we're rewriting: https://github.com/PostHog/posthog/blob/master/ee/clickhouse/queries/experiments/trend_experiment_result.py
- For Trend experiments, we store two queries: `count` and `exposure`
- For the `count` query, we just modify it adding the date range and feature flag breakdown
- For the exposure query, there are 3 possible cases:
	- If the `count` query uses some math aggregation (like `avg`), we need special handling. I found a potential issue with the current implementation - **this part is not ported yet, need to have a think on how to fix this / discuss with Neil**
	- Else if the exposure query is explicitly set, we modify this query in the same way as the `count` query
	- Else if the exposure query is not set, we default to counting `$feature_flag_called` events as our exposure

## How did you test this code?
Added tests

### CODE REVIEW

File: posthog/hogql_queries/experiment_trend_query_runner.py [hunk 1]
```diff
@@ -19,14 +28,141 @@ def __init__(self, *args, **kwargs):
...
+                        value=[self.feature_flag.key],
+                        operator="exact",
+                        type="event",
+                    ),
+                ],
+            )
+
+        return prepared_exposure_query
+
     def calculate(self) -> ExperimentTrendQueryResponse:
```
Comment (dmarticus) [inline posthog/hogql_queries/experiment_trend_query_runner.py:166]: this isn't in the scope of this PR, so feel free to ignore, but when reviewing this file I noticed that we could use a more threadsafe data structure to calculate the results, since we're running the queries in parallel. e.g., you could do this
```
def calculate(self) -> ExperimentTrendQueryResponse:
    count_response = None
    exposure_response = None
    errors = []
    result_queue = queue.Queue()

    def run(query_runner: TrendsQueryRunner, is_parallel: bool):
        try:
            result = query_runner.calculate()
            result_queue.put(result)
        except Exception as e:
            errors.append(e)
        finally:
            if is_parallel:
                from django.db import connection
                connection.close()

    # ... (rest of the method)

    # After joining threads
    try:
        count_response = result_queue.get_nowait()
        exposure_response = result_queue.get_nowait()
    except queue.Empty:
        pass

    # ... (rest of the method)
```

up to you though, not blocking.

File: frontend/src/queries/schema.ts [hunk 2]
```diff
@@ -1588,8 +1588,8 @@ export interface ExperimentFunnelQuery extends DataNode<ExperimentFunnelQueryRes
 
 export interface ExperimentTrendQuery extends DataNode<ExperimentTrendQueryResponse> {
     kind: NodeKind.ExperimentTrendQuery
-    count_source: TrendsQuery
-    exposure_source: TrendsQuery
+    count_query: TrendsQuery
+    exposure_query?: TrendsQuery // Defaults to $feature_flag_called if not specified
```
Comment (dmarticus) [inline frontend/src/queries/schema.ts:1592]: where is this default behavior specified?

**Review by dmarticus — APPROVED** [review]
a couple of questions, but this looks good overall!  I'm sorry to took me so long to get to it.

File: posthog/hogql_queries/experiment_trend_query_runner.py (hunk 1 above)
Comment (jurajmajerik) [reply posthog/hogql_queries/experiment_trend_query_runner.py]: Thanks for pointing this out! I don't think we necessarily need to use the queue here, but I did find another issue.

tldr; I was storing the result of the thread calculation incorrectly. The tests didn't surface this as we don't use threads in them.

I'm mimicking the implementation from here: https://github.com/PostHog/posthog/blob/master/posthog/hogql_queries/insights/trends/trends_query_runner.py#L300

I'm using a shared dict to store results. Each thread is assigned a distinct key ('count_response' or 'exposure_response') to store its result. Since the keys are unique, there's no chance of overwriting each other's results.

Let me know if I'm missing something!

File: frontend/src/queries/schema.ts (hunk 2 above)
Comment (jurajmajerik) [reply frontend/src/queries/schema.ts]: clarified the comment
